pkg/workflows/workflow.go

```diff
@@ -566,3 +582,35 @@ func (w *workflowLifecycle) injectActiveDeadlineSeconds(wf *unstructured.Unstruc
 
 	return nil
 }
+
+func IsValidV1WorkFlow(obj runtime.Object) error {
+	wf, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("error : unstructured type.")
+	}
+
+	if wf.GetKind() != "Workflow" || wf.GetAPIVersion() != "argoproj.io/v1alpha1" {
+		return fmt.Errorf("error : unsupported object from v1workflow update event %v", obj)
+	}
+	return nil
+}
+
+// extract addon-name and lifecyclestep from a workflow name string generated based on
+// api types
+func ExtractAddOnNameAndLifecycleStep(addonworkflowname string) (string, string, error) {
```

Comment:
Where is this method used, I don't see anywhere it's utilized I also don't get why it's needed since we will always know the format of the workflow names as we generate them from lifecycle step; there's already a `GetFormattedWorkflowName(lifecycleStep)` method on the Addon.
https://github.com/keikoproj/addon-manager/blob/f6e6c47744d9e29f8af113820764a3be4fa857af/api/addon/v1alpha1/addon_types.go#L361

Reply:
ExtractAddOnNameAndLifecycleStep is heavily used in PR 118

Reply:
https://github.com/keikoproj/addon-manager/pull/123/commits